### PR TITLE
feat(datasource-mongo): add support for ssh tunnelling

### PR DESCRIPTION
**Pull request by ghusse**
Linked to CU-86by94f32

### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/datasource-mongo/src/connection/create-connection.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    throw e;
+  }
+}
+
+export default async function createConnection(params: ConnectionParams): Promise<Connection> {
+  const { uri, connection } = params;
+  const { ssh, ...mongooseOptions } = connection ?? {};
+
+  if (ssh) {
+    return connectTroughSSH(params);
```
Comment (Scra3) [inline packages/datasource-mongo/src/connection/create-connection.ts:98]: `createMongooseConnectionThroughSsh`

File: packages/datasource-mongo/src/errors.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/* eslint-disable max-classes-per-file */
+import { BusinessError } from '@forestadmin/datasource-toolkit';
+
+export class SshConnectError extends BusinessError {
+  constructor(message: string, debugSshUri: string) {
+    super(
+      `Your ssh connection has encountered an error. ` +
```
Comment (Scra3) [inline packages/datasource-mongo/src/errors.ts:7]: Nice, it the same error than the DS sql 🧠

File: packages/datasource-mongo/src/connection/create-connection.ts (hunk 1 above)
Comment (ghusse) [reply packages/datasource-mongo/src/connection/create-connection.ts]: 👌

File: packages/datasource-toolkit/src/errors.ts [hunk 3]
```diff
@@ -7,18 +7,31 @@ export class BusinessError extends Error {
...
+export class IntrospectionFormatError extends BusinessError {
   constructor(sourcePackageName: '@forestadmin/datasource-sql' | '@forestadmin/datasource-mongo') {
     const message =
       `This version of introspection is newer than this package version. ` +
       `Please update ${sourcePackageName}`;
     super(message);
   }
+
+  /** @deprecated use name instead */
+  public get type() {
```
Comment (Scra3) [inline packages/datasource-toolkit/src/errors.ts:34]: Is it useful ?

File: packages/datasource-mongo/test/index.integration.test.ts [hunk 4]
```diff
@@ -52,12 +57,156 @@ describe('Datasource Mongo', () => {
...
+                  port: 2223,
+                  username: 'forest',
+                  privateKey,
+                },
+              },
+            }),
+          ).rejects.toThrow(SshConnectError);
+        });
+
+        it('should throw an error with the righ url', async () => {
```
Comment (Scra3) [inline packages/datasource-mongo/test/index.integration.test.ts:143]: right

Comment (ghusse) [inline packages/datasource-mongo/test/index.integration.test.ts:143]:
```suggestion
        it('should throw an error with the right url', async () => {
```

File: packages/datasource-mongo/docker-compose.yml [hunk 5]
```diff
@@ -8,4 +8,20 @@ services:
...
+      - PGID=1000
+      - TZ=Etc/UTC
+      - PUBLIC_KEY_FILE=/config/id_rsa.pub # path to the public key file inside the container
+      - USER_NAME=forest
+      - DOCKER_MODS=linuxserver/mods:openssh-server-ssh-tunnel # enable AllowTcpForwarding here
+      - LOG_STDOUT=true
+    volumes:
+      - ./test/ssh-config:/config
+    ports:
+      - '2223:2222'
```
Comment (Scra3) [inline packages/datasource-mongo/docker-compose.yml:27]: Can we use an other port? 
The ssh server on forestadmin-server uses the 2223 :/

Comment (ghusse) [reply packages/datasource-mongo/docker-compose.yml:27]: Sure, I'll change that

**Review by Scra3 — APPROVED** [review]

**Pull request merged**